### ISSUE TITLE
Please provide a demo

### ISSUE THREAD

**Issue by AyabongaQwabi** [ikhokha/ik-pay-api-examples#9]
The current Node.js API example provided in the documentation is somewhat limited. It primarily focuses on the request body structure, leaving out crucial details on how to make the actual HTTP request and handle potential responses.

To enhance the clarity and usability of the example, I suggest the following improvements:

1. Complete Code Example:

Provide a full code example demonstrating how to make a POST request to the specified endpoint using either axios or the native fetch API.

Include code snippets for:

- Installing necessary dependencies (axios or node-fetch)

- Constructing the request body based on the provided example

- Making the HTTP POST request to the appropriate endpoint

- Handling both successful and error responses

2. Response Handling:

Clearly document the expected response formats for both successful and error scenarios.
Provide examples of potential error responses, such as:

- 400 Bad Request

- 401 Unauthorized

- 403 Forbidden

- 500 Internal Server Error

- 503 Service Unavailable

Explain how to parse the response body to extract relevant information.

To enhance the clarity and usability of the example, I suggest providing a comprehensive code example demonstrating the entire process e.g:

```
const axios = require('axios');

const apiKey = 'YOUR_API_KEY';
const secretKey = 'YOUR_SECRET_KEY';

const requestBody = {
    // ... your request body as specified in the documentation
};

const config = {
    headers: {
        'Content-Type': 'application/json',
        'Authorization': 'Basic ' + Buffer.from(apiKey + ':' + secretKey).toString('base64')
    }
};

axios.post('https://api.ikhokha.com/v1/payments', requestBody, config)
    .then(response => {
        console.log('Payment successful:', response.data);
    })
    .catch(error => {
        console.error('Payment failed:', error.response.data);

        // Handle specific error codes:
        if (error.response.status === 401) {
            console.error('Unauthorized: Please check your API credentials.');
        } else if (error.response.status === 500) {
            console.error('Internal Server Error: Please contact Ikhokha support.');
        } else {
            console.error('An unexpected error occurred:', error);
        }
    });
```

By incorporating these suggestions, the Node.js API example will become significantly more valuable to developers, streamlining the integration process and reducing potential issues.

**Comment by wiredmartian** [comment]
Hey @AyabongaQwabi 

Forgive me, but this suggestions feels like something regurgitate from AI. The Nodejs example does in fact show you how to make an HTTP request (https://github.com/ikhokha/ik-pay-api-examples/blob/b5e60e9dc8f78c89e38649530929fbd58b973ab6/nodejs/pay-api-app/backend/paylink.js#L14), constructing the body, handling response, etc. All these are **working** examples.

Here's a demo video based on the exact examples: https://www.youtube.com/watch?v=qaUl2P70734

It'll explain some parts you might feel are missing

**Comment by AyabongaQwabi** [comment]
Yes, that text was produced using AI because it effectively enhances and articulates my points. Your API fails to provide essential details, such as the meanings of HTTP error status codes and crucial information about HTTP payload properties. For example, the property "mode" needs clarification. What does it entail, how many modes are there, and can you provide specific examples?

Additionally, I need a clear explanation of the callback URL—what it means and its purpose.

Moreover, your examples are not beginner-friendly, as I clearly stated in my previous message. What specific installations are required? Where exactly should I place the code? How can I integrate this API with modern libraries like React or Next.js?

Your examples need to offer much more depth.

This text was also improved with AI.

**Comment by AyabongaQwabi** [comment]
siselapho i regurgitate ngu Yihlo kwedini ndithi ndithetha nawe kakhle ufike ngesgezo, ziqaqadekile ezinto zakwa Ikhoka sies

**Comment by AyabongaQwabi** [comment]
Tshini lentwana ndithi ndibhala kakhle even to the point yoba ndisebenzise i AI ukwenzela ndivakale kakhle ise ifike yona ithi ndibhale umgabho yemadoda
